specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
+has handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                           | Retry | Delivery completed | Error |
+| ------------- | --------------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply)       | No    | Yes                | No    |
+| `202`         | Event accepted                    | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `400`         | Unparsable event                  | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist           | Yes   | No                 | Yes   |
+| `409`         | Conflict / Processing in progress | Yes   | No                 | Yes   |
+| `429`         | Too Many Requests / Overloaded    | Yes   | No                 | Yes   |
+| other `4xx`   | Error                             | No    | No                 | Yes   |
+| other `5xx`   | Error                             | Yes   | No                 | Yes   |
+
+\* Unspecified `1xx`, `2xx`, and `3xx` response codes are **reserved for future
+extension**. Event recipients SHOULD NOT send these response codes in this spec
+version, but event senders MUST handle these response codes as errors and
+implement appropriate failure behavior.
+
+Recipients MUST be able to handle duplicate delivery of events (for example, via
+idempotency or tracking event delivery state) and MUST accept delivery of
+duplicate events, as the event acknowledgement could have been lost in return to
+the sender. As specified in the
+[CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/primer.md#id),
+event recipients MUST use the
+[`source` and `id` attributes](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#required-attributes)
+to determine duplicate events if needed. This specification does not describe
+state requirements for clients which need to detect duplicate events. (see
+[observability](#observability) for an example case where other event attributes
+may vary from one delivery attempt to another).
+
+Where possible, event senders SHOULD re-attempt delivery of events where the
+HTTP request failed or returned a retryable status code. It is RECOMMENDED that
+event senders implement some form of congestion control (such as exponential
+backoff) when managing retry timing. This specification does not document any
+specific congestion control algorithm or parameters.
+[Brokers](./overview.md#broker) and [Channels](./overview.md#channel) MUST
```

Comment:
Like the previous comment, MUST may be too strong here. I don't want my broker/channel to spam my ksvc if the error returned by the ksvc is error and not going to change no matter how many times it is called.

Reply:
Added:

> Congestion control MAY cause event delivery to fail or retry attempts to be skipped

(Delivery failure would trigger dead-letter behavior, see #25 for details of that.)